great_expectations/cli/datasource.py

```diff
@@ -147,23 +148,151 @@ def add_datasource(context):
     return data_source_name
 
 
-def profile_datasource(context, data_source_name, max_data_assets=20):
+def profile_datasource(context, data_source_name, data_assets=None, profile_all_data_assets=False, max_data_assets=20):
     """"Profile a named datasource using the specified context"""
+
+    msg_intro = """
+========== Profiling ==========
+
+Profiling '{0:s}' will create candidate expectations and documentation.
+
+Please note: Profiling is still a beta feature in Great Expectations.  The current profiler will evaluate the entire 
+data source (without sampling), which may be very time consuming. 
+As a rule of thumb, we recommend starting with data smaller than 100MB.
```

Comment:
Since "100MB" is meaningless to most non-pandas datasets, let's just change this to "we recommend starting with a small dataset." so that small is in the eye of the beholder (and their knowledge of their own compute)